src/answerer.rs

```diff
@@ -0,0 +1,355 @@
+    defillama_http_client: Arc<HttpClient>,
+    mut active_oracle: models::ActiveOracle,
+) -> anyhow::Result<()> {
+    if let Some(tx_hash) = active_oracle.answer_tx_hash {
+        tracing::warn!(
+            "answering procedure already active for oracle with tx hash 0x{:x}, skipping",
+            tx_hash.0
+        );
+        return Ok(());
+    }
+
+    match is_active_oracle_expired(
+        db_connection_pool.clone(),
+        signer.clone(),
+        &mut active_oracle,
+    )
+    .await
+    {
+        Ok(expired) => {
+            if expired {
+                let mut db_connection = match db_connection_pool
+                    .get()
+                    .context("could not get new connection from pool")
+                {
+                    Ok(db_connection) => db_connection,
+                    Err(error) => {
+                        tracing::error!(
+                    "could not get database connection while trying to update oracle's answer tx hash: {:#}",
+                    error
+                );
+                        return Ok(());
+                    }
+                };
+
+                tracing::warn!("oracle is expired, skipping and deleting");
+                if let Err(error) = active_oracle.delete(&mut db_connection) {
+                    tracing::error!("{:#}", error);
+                }
+                return Ok(());
+            }
+        }
+        Err(error) => {
+            tracing::error!("could not get expiration status for oracle: {:#}", error);
+            return Ok(());
+        }
+    }
+
+    let answer = match &active_oracle.answer {
+        Some(answer) => {
+            tracing::info!("reusing saved answer {}", answer.0);
+            Some(answer.0)
+        }
+        None => {
+            let answer =
+                specification::answer(&active_oracle.specification, defillama_http_client).await;
+            if let Some(answer) = answer {
+                let mut db_connection = match db_connection_pool
+                    .get()
+                    .context("could not get new connection from pool")
+                {
+                    Ok(db_connection) => db_connection,
+                    Err(error) => {
+                        tracing::error!(
+                    "could not get database connection while trying to update oracle's answer: {:#}",
+                    error
+                );
+                        return Ok(());
+                    }
+                };
+
+                if let Err(error) = active_oracle.update_answer(&mut db_connection, answer) {
+                    tracing::error!("{:#}", error);
+                    return Ok(());
+                }
+            }
+            answer
+        }
+    };
+    if let Some(answer) = answer {
+        // if we arrive here, an answer is available and we should submit it
+
+        tracing::info!("answering with value {}", answer);
```

Comment:
Maybe we could add the oracle address in the logs here. and below too, no?

Reply:
In here we already have the oracle address in the logs in the spans section thanks to `instrument(info_span!("answer", chain_id, oracle_address))` on the answering task future.